### PR TITLE
Webpack Support

### PR DESCRIPTION
**Pull request by andrewjtait**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+# 4.0.0
+
+## Webpack Support
+
+Carbon was previously designed around a Browserify based build, we have now migrated to explicitly support Webpack. We have made this decision based on the overwhelming preference in the JavaScript community to work with Webpack, while over the past couple of years support for Browserify has dropped and the library has not been able to keep up with the pace of Webpack. Switching to Webpack also provides us with all the latest development tools such as faster recompilation, hot reloading, and code splitting.
+
+We have updated several components to import assets through their JavaScript file. Due to this change we now require Webpack to precompile the components.
+
+If you use carbon-factory, you can upgrade to [v4.1.0](https://github.com/Sage/carbon-factory/releases/tag/v4.0.1) which provides preconfigured Webpack support for working with the Carbon library. If you prefer to manage your own Webpack build, we currently rely on the [Parcelify Loader](https://www.npmjs.com/package/parcelify-loader). We will be looking to drop this requirement in the future.
```
Comment (SeanArmstrong) [inline CHANGELOG.md]: Incorrect link

**Comment by SeanArmstrong** [comment]
Dev tested 👍

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module.exports = require('carbon-factory/webpack.config')({
+  entryPoint: '/demo/main.js',
```
Comment (SeanArmstrong) [inline webpack.config.js]: The port for the demo site was `8095`. This has now changed to `8080`

File: demo/site-map.js [hunk 3]
```diff
@@ -47,15 +47,15 @@ export default new SiteMapHelper({
...
-      'rails-part-3:-updating-data': `${tutorialsPath}/carbon-rails/updating-data.md`
-    }
-  }
+  // '/tutorials': {
+  //   items: {
+  //     'rails-part-1:-hello-world': `${tutorialsPath}/carbon-rails/hello-world.md`,
+  //     'rails-part-2:-introducing-data': `${tutorialsPath}/carbon-rails/introducing-data.md`,
+  //     'rails-part-3:-updating-data': `${tutorialsPath}/carbon-rails/updating-data.md`
+  //   }
+  // }
```
[the remark this answers is not in the document]
Comment (andrewjtait) [reply demo/site-map.js]: I've disabled this for now as it is outdated documentation that we shouldn't really be promoting. We will need to go back to update with new guides when we have time.

File: webpack.config.js (hunk 2 above)
Comment (andrewjtait) [reply webpack.config.js]: updated any reference of 8095 to use 8080

**Comment by iliaoni** [comment]
Sanity tested on Safari, Chrome, Firefox, IE11 on W8.1, Edge on Win10.

No issues except the date dropdown CARBON-1077.

File: package.json [hunk 4]
```diff
@@ -31,10 +35,9 @@
   },
   "devDependencies": {
     "babel-standalone": "~6.17.0",
-    "carbon-factory": "3.3.3",
+    "carbon-factory": "git://github.com/Sage/carbon-factory.git#build-webpack",
```
Comment (ianoxley) [inline package.json]: Can this be updated yet to point to a non-branch version?

Comment (andrewjtait) [reply package.json]: ah yes it can

**Pull request merged**